src/ray/core_worker/task_event_buffer.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2022 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "ray/common/asio/instrumented_io_context.h"
+#include "ray/common/id.h"
+#include "ray/common/task/task_spec.h"
+#include "src/ray/protobuf/common.pb.h"
+#include "src/ray/protobuf/gcs.pb.h"
+
+namespace ray {
+namespace core {
+
+namespace worker {
+
+/// An interface for a buffer that stores task status changes and profiling events,
+/// and reporting these events to the GCS periodically.
+///
+/// Dropping of task events
+/// ========================
+/// Task events will be lost in the below cases for now:
+///   1. If any of the gRPC call failed, the task events will be silently dropped. This
```

Comment:
Could we log an error / warning message?

Reply:
Oh right yes, not so silent.